src/core/shape_inference/include/slice_shape_inference_utils.hpp

```diff
@@ -115,6 +115,12 @@ U get_value_or_limit_of(const element::Type_t& type, const T& value) {
 namespace op {
 namespace slice {
 
+#if INTPTR_MAX == INT64_MAX
+using data_type = int64_t;
+#else
+using data_type = int32_t;
+#endif
```

Comment:
why do we need different data types here? our expectation is that OpenVINO opset should be 32/64 bits independent.

Could you please provide more details about each case?

Reply:
The original issues are below:

```
[ RUN      ] SliceStaticShapeInferenceTest.reverse_steps_start_stop_outside_dimension_default_axes
/home/ov2022/workspace/wasm/openvino/src/plugins/intel_cpu/tests/unit/shape_inference_test/slice_shape_inference_test.cpp:40: Failure
Expected equality of these values:
  output_shapes.front()
    Which is: { 3, 2, 5, 4294967289, 0 }
  StaticShape({3, 2, 5, max_d, 3})
    Which is: { 3, 2, 5, 4294967295, 3 }
[  FAILED  ] SliceStaticShapeInferenceTest.reverse_steps_start_stop_outside_dimension_default_axes (1 ms)
[ RUN      ] SliceStaticShapeInferenceTest.forward_step_all_data_in_const_map
/home/ov2022/workspace/wasm/openvino/src/plugins/intel_cpu/tests/unit/shape_inference_test/slice_shape_inference_test.cpp:107: Failure
Expected equality of these values:
  output_shapes.front()
    Which is: { 10, 3, 0, 4, 1073741823, 2147483645, 3 }
  StaticShape({10, 3, 0, 4, max_d, max_d, 3})
    Which is: { 10, 3, 0, 4, 4294967295, 4294967295, 3 }
[  FAILED  ] SliceStaticShapeInferenceTest.forward_step_all_data_in_const_map (1 ms)
```

In 32bits platform, max_d is 0xffffffff
https://github.com/openvinotoolkit/openvino/blob/4fd38844a280c64346e229aaf42b12c0ded2847b/src/plugins/intel_cpu/tests/unit/shape_inference_test/slice_shape_inference_test.cpp#L22

so the input_shape will be: {3, 4, 5, 0xffffffff, 0xffffffff}
https://github.com/openvinotoolkit/openvino/blob/4fd38844a280c64346e229aaf42b12c0ded2847b/src/plugins/intel_cpu/tests/unit/shape_inference_test/slice_shape_inference_test.cpp#L34

It will cause is_norm_dim_max is false due to dim.get_min_length() is 64bit data type
https://github.com/openvinotoolkit/openvino/blob/4fd38844a280c64346e229aaf42b12c0ded2847b/src/core/shape_inference/include/slice_shape_inference_utils.hpp#L137

then (lb, ub) computation will goto another path and get incorrect value.
https://github.com/openvinotoolkit/openvino/blob/4fd38844a280c64346e229aaf42b12c0ded2847b/src/core/shape_inference/include/slice_shape_inference_utils.hpp#L160-L161

Reply:
A simple method is:
`const auto is_norm_dim_max = ov::internal::is_max(size_t(norm_dim));`